specification/network/resource-manager/Microsoft.Network/stable/2019-12-01/firewallPolicy.json

```diff
@@ -818,10 +828,63 @@
           "items": {
             "type": "string"
           }
+        },
+        "sourceIpGroups": {
+          "type": "array",
+          "description": "List of source IpGroups for this rule.",
+          "items": {
+            "type": "string"
+          }
```

Comment:
adding new properties is a breaking change.
Firstly, it will break existing SDKs.. (if it fails due to additional properties).

And what is the behavior at the server side, when user does a PUT with old SDK (without new fields) on a resource that has these new fields set.

Reply:
How will adding new properties be breaking change? A PUT with old SDK will result in a new model with these fields as empty List. What is the guidance for adding new properties?

Reply:
` [DataMember]
        public List<string> SourceIpGroups
        {
            get
            {
                if (this.sourceIpGroups == null)
                {
                    this.sourceIpGroups = new List<string>();
                }

                return this.sourceIpGroups;
            }

            set
            {
                // Trim white spaces from user input
                this.sourceIpGroups = value.Select(sourceIpGroup => sourceIpGroup.Trim()).ToList();
            }
        }
```

Reply:
@sanjaiganesh this PR is targeting NRP's release branch for new API version which isn't available to the customer yet and isn't in the master. This shouldn't be considered breaking change.

Reply:
New property added to response 

If a new property/field is added to the response an API, the GET-PUT pipeline will be broken. Consider the case where from portal a customer updates the value of a new property "A". Another customer does a GET of this resource using the SDK. The SDK will ignore the property since it does not understand it. From the SDK, the customer does a PUT using the model that was returned from the GET. This will overwrite the change made by the first customer from the portal.  

---
In reply to: [388080643](https://github.com/Azure/azure-rest-api-specs/pull/8477#discussion_r388080643) [](ancestors = 388080643)